Source/WebKit/UIProcess/ProcessThrottler.cpp

```diff
@@ -317,4 +317,37 @@ void ProcessThrottler::TimedActivity::updateTimer()
         m_timer.startOneShot(m_timeout);
 }
 
+template <typename T>
+static Vector<String> activityNames(const T& activities)
+{
+    Vector<String> names;
+    for (const auto* activity : activities) {
+        if (activity)
```

Comment:
You may want to skip activities that return true for `isQuietActivity()` too. Those don't have a name and are used for very verbose things like IPC.